packages/apollo-gateway/src/__tests__/gateway/executor.test.ts

```diff
@@ -35,4 +36,24 @@ describe('ApolloGateway executor', () => {
       'Variable "$first" got invalid value "3"; Expected type Int.',
     );
   });
+
+  it('still sets the ApolloServer executor on load rejection', async () => {
+    jest.spyOn(console, 'error').mockImplementation();
+
+    const gateway = new ApolloGateway({
+      // Empty service list will throw, which is what we want.
+      serviceList: [],
+    });
+
+    const server = new ApolloServer({
+      gateway,
+      subscriptions: false,
+    });
+
+    // Ensure the throw happens to maintain the correctness of this test.
+    await expect(
+      server.executeOperation({ query: '{ __typename }' })).rejects.toThrow();
+
+    expect(server.requestOptions.executor).toStrictEqual(gateway.executor);
```

Comment:
I think you want referential identity here, though this probably works out anyway.
```suggestion
    expect(server.requestOptions.executor).toBe(gateway.executor);
```